### DISCUSSION TITLE
i have idea to compiile tailwind in php in production mode

### DISCUSSION THREAD

**Discussion by vahidalvandi** [gregsullivan/_tw#123] [General]
thank you for _tw , i use [arnolem/tailwindphp](https://github.com/arnolem/tailwindphp) in other php project , this can add new feature to allow re-compile theme online whitoit npm and just with php .

there is sample code https://github.com/tailwindlabs/tailwindcss/discussions/8105

**Comment by gregsullivan** [comment]
Hi Vahid—thanks for posting this!

I definitely understand the motivation here, but it runs counter to my goals for _tw. I'm specifically interested in:

* Developing locally, building a CSS file, and having that CSS file act as the source of the site's CSS in its entirety
* Not adding any dependencies to the production server
* Not accommodating classes being added directly to the block editor

This is obviously opinionated, but I do think the cons as listed as your second link are worth considering: I'm not comfortable with a slow cache-warming period upon deployment, and I think requiring support for `shell-exec` and the ability to run a CLI binary in production would be pretty daunting to overcome.

I think this is a great idea, and I get why it would be great for some people, but unfortunately it doesn't fit with my current and future plans for what I think _tw should be.

Thanks again for suggesting this, and I'll convert this issue to a discussion in the meantime!

**Comment by vahidalvandi** [comment]
I worked with this method in Laravel and it had a good result. It is the same as wp. It checks a specific path and converts the css file from the html content of the template to a css file.   Something like compile scss in bootstrap formats, which is normal.  But it can be interesting for tw because when we want to develop the template using tw snips, it is enough to put a piece of html code in the prod environment and then compile it online with php and run the css file.   The second link I sent may not be very useful, the first link is important
